azure-spring-boot-samples/azure-storage-spring-boot-sample/src/main/java/sample/storage/StorageService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package sample.storage;
+
+import com.microsoft.azure.storage.blob.BlobRange;
+import com.microsoft.azure.storage.blob.BlockBlobURL;
+import com.microsoft.azure.storage.blob.ContainerURL;
+import com.microsoft.azure.storage.blob.TransferManager;
+import com.microsoft.azure.storage.blob.models.ContainerCreateResponse;
+import com.microsoft.rest.v2.RestException;
+import com.microsoft.rest.v2.util.FlowableUtil;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+@SuppressFBWarnings({"RV_RETURN_VALUE_IGNORED"})
+public class StorageService {
+    public static void uploadFile(BlockBlobURL blob, File sourceFile) throws IOException {
+        logInfo("Start uploading file %s...", sourceFile);
+        final AsynchronousFileChannel fileChannel = AsynchronousFileChannel.open(sourceFile.toPath());
+
+        TransferManager.uploadFileToBlockBlob(fileChannel, blob, 8 * 1024 * 1024, null)
```

Comment:
suggest to convert to completable, as subscribe should run in another thread.

Reply:
Thanks, updated.